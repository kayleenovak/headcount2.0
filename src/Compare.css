.compare-container {
  display: flex;
  justify-content: center;
}