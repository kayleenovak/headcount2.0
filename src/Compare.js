import React from 'react';
import CompareCard from './CompareCard.js';
import StatsCard from './StatsCard.js'
import './Compare.css';

const CompareContainer = (props) => {
  return (
    <div class='compare-container'>
      <CompareCard removeComparedSchool={ props.removeComparedSchool } school={ props.schoolsForComparison[0] } schoolStats={ props.comparedSchools[props.schoolsForComparison[0]].stats } />
      <StatsCard schoolsForComparison={ props.schoolsForComparison } comparison={ props.comparison }/>

      {
        props.schoolsForComparison.length > 1 ? <CompareCard removeComparedSchool={ props.removeComparedSchool } school={ props.schoolsForComparison[1] } schoolStats={ props.comparedSchools[props.schoolsForComparison[1]].stats } /> : null
      }
    </div>
  )
}

export default CompareContainer

