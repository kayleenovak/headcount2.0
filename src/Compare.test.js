import React from 'react';
import { shallow, mount } from 'enzyme';
import Compare from './Compare.js';

describe('Compare', () => {
  it('should match the snapshot', () => {
    const mockSchools = ['COLORADO', 'COLORADO SPRINGS 11']
    const mockCompare = {
      'COLORADO': {
        'location': 'COLORADO',
        'stats': {
          '2004': 1
        }
      },
      'COLORADO SPRINGS 11': {
        'location': 'COLORADO SPRINGS 11',
        'stats': {
          '2004': 1
        }
      }
    }
    const mockFunction = jest.fn()
    const mockComparison = {
      'COLORADO': .4,
      'COLORADO SPRINGS 11': .4,
      'compared': 1
    }
    const wrapper = shallow(<Compare schoolsForComparison={ mockSchools } comparedSchools={ mockCompare } removeComparedSchool={ mockFunction } comparison={ mockComparison }/>)

    expect(wrapper).toMatchSnapshot()
  })
})