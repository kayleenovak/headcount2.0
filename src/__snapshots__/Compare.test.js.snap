// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    comparedSchools={
        Object {
            "COLORADO": Object {
              "location": "COLORADO",
              "stats": Object {
                "2004": 1,
              },
            },
            "COLORADO SPRINGS 11": Object {
              "location": "COLORADO SPRINGS 11",
              "stats": Object {
                "2004": 1,
              },
            },
          }
    }
    comparison={
        Object {
            "COLORADO": 0.4,
            "COLORADO SPRINGS 11": 0.4,
            "compared": 1,
          }
    }
    removeComparedSchool={[Function]}
    schoolsForComparison={
        Array [
            "COLORADO",
            "COLORADO SPRINGS 11",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CompareCard
          removeComparedSchool={[Function]}
          school="COLORADO"
          schoolStats={
                    Object {
                              "2004": 1,
                            }
          }
/>,
        <StatsCard
          comparison={
                    Object {
                              "COLORADO": 0.4,
                              "COLORADO SPRINGS 11": 0.4,
                              "compared": 1,
                            }
          }
          schoolsForComparison={
                    Array [
                              "COLORADO",
                              "COLORADO SPRINGS 11",
                            ]
          }
/>,
        <CompareCard
          removeComparedSchool={[Function]}
          school="COLORADO SPRINGS 11"
          schoolStats={
                    Object {
                              "2004": 1,
                            }
          }
/>,
      ],
      "class": "compare-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "removeComparedSchool": [Function],
          "school": "COLORADO",
          "schoolStats": Object {
            "2004": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparison": Object {
            "COLORADO": 0.4,
            "COLORADO SPRINGS 11": 0.4,
            "compared": 1,
          },
          "schoolsForComparison": Array [
            "COLORADO",
            "COLORADO SPRINGS 11",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "removeComparedSchool": [Function],
          "school": "COLORADO SPRINGS 11",
          "schoolStats": Object {
            "2004": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CompareCard
            removeComparedSchool={[Function]}
            school="COLORADO"
            schoolStats={
                        Object {
                                    "2004": 1,
                                  }
            }
/>,
          <StatsCard
            comparison={
                        Object {
                                    "COLORADO": 0.4,
                                    "COLORADO SPRINGS 11": 0.4,
                                    "compared": 1,
                                  }
            }
            schoolsForComparison={
                        Array [
                                    "COLORADO",
                                    "COLORADO SPRINGS 11",
                                  ]
            }
/>,
          <CompareCard
            removeComparedSchool={[Function]}
            school="COLORADO SPRINGS 11"
            schoolStats={
                        Object {
                                    "2004": 1,
                                  }
            }
/>,
        ],
        "class": "compare-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "removeComparedSchool": [Function],
            "school": "COLORADO",
            "schoolStats": Object {
              "2004": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "comparison": Object {
              "COLORADO": 0.4,
              "COLORADO SPRINGS 11": 0.4,
              "compared": 1,
            },
            "schoolsForComparison": Array [
              "COLORADO",
              "COLORADO SPRINGS 11",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "removeComparedSchool": [Function],
            "school": "COLORADO SPRINGS 11",
            "schoolStats": Object {
              "2004": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
